libkmod/libkmod.c

```diff
@@ -490,9 +502,14 @@ int kmod_lookup_alias_from_builtin_file(struct kmod_ctx *ctx, const char *name,
 		/* already mark it as builtin since it's being created from
 		 * this index */
 		kmod_module_set_builtin(mod, true);
-		*list = kmod_list_append(*list, mod);
-		if (*list == NULL)
+		node = kmod_list_append(*list, mod);
+		if (node == NULL) {
+			ERR(ctx, "out of memory\n");
+			kmod_module_unref(mod);
 			return -ENOMEM;
+		} else {
```

Comment:
Nit: drop the else-after-return, similar to how we don't have else-after-goto in this patch.

Reply:
Adjusted